packages/grpc-js/src/make-client.ts

```diff
@@ -49,9 +50,9 @@ export interface MethodDefinition<RequestType, ResponseType>
   extends ClientMethodDefinition<RequestType, ResponseType>,
     ServerMethodDefinition<RequestType, ResponseType> {}
 
-export interface ServiceDefinition {
+export type ServiceDefinition<ImplementationType = UntypedServiceImplementation> = {
```

Comment:
Why change `interface` to `type` here?

Also, I don't understand how this would be used. Can you provide an example of using this type definition with and without the type parameter?

Reply:
A `type` is needed, otherwise the typescript compiler throws an error for `interface`:
> A computed property name must be of type 'string', 'number', 'symbol', or 'any'.ts(2464)

See this PR https://github.com/microsoft/TypeScript/pull/26797 

If I generate code from my proto files via  `grpc_tools_node_protoc_ts`, I get these service definitions:
```ts
interface IMyCustomService extends grpc.ServiceDefinition<grpc.UntypedServiceImplementation> {
  get: IMyCustomService_IGet;
}

interface IMyCustomService_IGet extends grpc.MethodDefinition<myservice_pb.GetRequest, google_protobuf_empty_pb.Empty> {
  prop: boolean; // false
  propDeserialize: grpc.deserialize<google_protobuf_empty_pb.Empty>;
}
```

I just looked at https://github.com/grpc/grpc-node/blob/4823d97aba704886a0eff755ff78383338cb6fe6/packages/grpc-native-core/index.d.ts#L315-L317 and added the default ImplementationType to be compatible to the interface used before.

Reply:
I did not remember that we had that definition in the other library. In that case, if the tests pass, I'm fine with this change.